### DISCUSSION TITLE
graphscope运行环境在k8s上初始化时间

### DISCUSSION THREAD

**Discussion by lvf0052** [alibaba/GraphScope#2325] [General]
![image](https://user-images.githubusercontent.com/18576873/206981900-101a4f0f-b5ac-4ce4-9fd7-1102600bb75c.png)

![image](https://user-images.githubusercontent.com/18576873/206981945-d20d3a17-3657-417b-b4eb-6d15d5ce8149.png)

问题描述：
我们在k8s上集群部署graphscope，使用graphscope的GLE引擎进行【在线的图计算】
每次进行图计算之前，都需要初始化一个graphscope【默认配置】的运行环境，时间在1分钟左右(未进行图数据的加载)，请问这个时间是否可以缩短，或者说是在日常使用的过程中，初始化运行环境的确是要一分钟左右。(初始化流程如上图所示)
在使用过程当中，各个环节所花时间如下：
构建计算环境时间：58s
数据加载时间：100s
图计算时间：1s
1. 数据加载的时间上已有优化思路，是代码的问题。
2. 构建计算环境时间这个暂时无优化思路。

**Comment by sighingnow** [comment]
> 每次进行图计算之前，都需要初始化一个graphscope【默认配置】的运行环境，时间在1分钟左右(未进行图数据的加载)，请问这个时间是否可以缩短，或者说是在日常使用的过程中，初始化运行环境的确是要一分钟左右。(初始化流程如上图所示)

这个时间包含拉起k8s 资源的时间么

**Reply by lvf0052** [reply]
> > 每次进行图计算之前，都需要初始化一个graphscope【默认配置】的运行环境，时间在1分钟左右(未进行图数据的加载)，请问这个时间是否可以缩短，或者说是在日常使用的过程中，初始化运行环境的确是要一分钟左右。(初始化流程如上图所示)
> 
> 这个时间包含拉起k8s 资源的时间么

感谢你的解答。
是的，这第一个步骤：【构建计算环境时间】就是包括了k8s拉起环境的时间，就是这句代码：g.session()
1)	系统在指定的命名空间下创建集群角色，并将集群该集群角色赋予创建该集群的用户
2)	若是首次执行g.session()方法，若k8s中不存在graphscope镜像，系统将去远程仓库进行进行拉取
3)	启动协调节点，该协调节点主要负责管理计算引擎跟etcd
4)	协调节点启动etcd，etcd主要配置共享和服务的注册和发现
5)	协调节点启动计算引擎及其vineyard，计算引擎负责算法的计算，vineyard负责数据的存储

58秒包括内容为：拉起K8s资源的时间(本地仓库里已经有镜像了)，不包括数据加载的时间和计算的时间。
